recipe/conda_build_config.yaml

```diff
@@ -165,6 +165,7 @@ zip_keys:
     - cxx_compiler_version      # [unix]
     - fortran_compiler_version  # [unix]
     - c_stdlib_version          # [unix]
+    - MACOSX_DEPLOYMENT_TARGET  # [osx]
```

Comment:
There's no need to put the `c_stdlib_version` (and related keys) into this zip key.
This tight coupling is only needed for the CUDA stuff.
We should do something like
```diff
+  -                             # [osx]
+    - c_stdlib_version          # [osx]
+    - MACOSX_DEPLOYMENT_TARGET  # [osx]
+# For CUDA, c_stdlib_version/cdt_name is zipped below with the compilers.
+  -                             # [linux and os.environ.get("CF_CUDA_ENABLED", "False") != "True"]
+    - c_stdlib_version          # [linux and os.environ.get("CF_CUDA_ENABLED", "False") != "True"]
+    - cdt_name                  # [linux and os.environ.get("CF_CUDA_ENABLED", "False") != "True"]
   -                             # [unix]
     - c_compiler_version        # [unix]
     - cxx_compiler_version      # [unix]
     - fortran_compiler_version  # [unix]
-    - c_stdlib_version          # [unix]
-    - MACOSX_DEPLOYMENT_TARGET  # [osx]
+    - c_stdlib_version          # [linux and os.environ.get("CF_CUDA_ENABLED", "False") == "True"]
     - cdt_name                  # [linux and os.environ.get("CF_CUDA_ENABLED", "False") == "True"]
     - cuda_compiler             # [linux and os.environ.get("CF_CUDA_ENABLED", "False") == "True"]
     - cuda_compiler_version     # [linux and os.environ.get("CF_CUDA_ENABLED", "False") == "True"]
     - docker_image              # [linux and os.environ.get("CF_CUDA_ENABLED", "False") == "True" and os.environ.get("BUILD_PLATFORM", "").startswith("linux-")]
```
or
```diff
+# For CUDA, c_stdlib_version/cdt_name is zipped below with the compilers.
+  -                             # [osx or (linux and os.environ.get("CF_CUDA_ENABLED", "False") != "True")]
+    - c_stdlib_version          # [osx or (linux and os.environ.get("CF_CUDA_ENABLED", "False") != "True")]
+    - MACOSX_DEPLOYMENT_TARGET  # [osx]
+    - cdt_name                  # [linux and os.environ.get("CF_CUDA_ENABLED", "False") != "True"]
   -                             # [unix]
     - c_compiler_version        # [unix]
     - cxx_compiler_version      # [unix]
     - fortran_compiler_version  # [unix]
-    - c_stdlib_version          # [unix]
-    - MACOSX_DEPLOYMENT_TARGET  # [osx]
+    - c_stdlib_version          # [linux and os.environ.get("CF_CUDA_ENABLED", "False") == "True"]
     - cdt_name                  # [linux and os.environ.get("CF_CUDA_ENABLED", "False") == "True"]
     - cuda_compiler             # [linux and os.environ.get("CF_CUDA_ENABLED", "False") == "True"]
     - cuda_compiler_version     # [linux and os.environ.get("CF_CUDA_ENABLED", "False") == "True"]
     - docker_image              # [linux and os.environ.get("CF_CUDA_ENABLED", "False") == "True" and os.environ.get("BUILD_PLATFORM", "").startswith("linux-")]
```
(Sorry, GitHub UI didn't allow me to put this into a proper `suggestion` since it touches things outside of this "hunk".)

Reply:
I see. Let's make a new PR for that. We've had trouble with these zips and proper rendering. It'll be easier to test as two steps.

Reply:
If you want.. That PR would then be a requisite for this PR then; i.e., we should not zip  `MACOSX_DEPLOYMENT_TARGET` with the compiler versions in any case.

Reply:
They are one and the same on osx. The macosx deployment target being set explicitly will be deprecated behavior after stdlib is done. The zipping here is to allow for a transition period as far as I understand it.